### PR TITLE
SP-307 - Chatty Messaging

### PR DESCRIPTION
**Pull request by gopipatell**

### CODE REVIEW

File: src/Purchase.js [hunk 1]
```diff
@@ -633,3 +633,22 @@ export class AuthCodeAdvice
         return new Message(RequestIdHelper.Id("authad"), Events.AuthCodeAdvice, data, true);
     }
 }
+
+export class TransactionUpdate
+{
+    constructor(m) 
+    {
+        console.log('TransactionUpdate m ', m);
```
Comment (RaymondK-mx51) [inline src/Purchase.js:641]: Does this line get triggered at all?

File: src/Spi.js [hunk 2]
```diff
@@ -1889,6 +1899,9 @@ class Spi {
             case Events.PrintingResponse:
                 this._handlePrintingResponse(m);
                 break;
+            case Events.TransactionUpdateMessage:
+                this._handleTransactionUpdateMessage(m);
```
Comment (RaymondK-mx51) [inline src/Spi.js:1903]: Does this line get triggered when debugging?

File: src/Purchase.js (hunk 1 above)
Comment (gopipatell) [reply src/Purchase.js]: I removed the console log

File: src/Spi.js (hunk 2 above)
Comment (gopipatell) [reply src/Spi.js:1904]: I was not triggered before but now it is

File: src/Spi.js [hunk 3]
```diff
@@ -1443,16 +1443,24 @@ class Spi {
...
+    }
+
     _handlePrintingResponse(m)
     {
         if (typeof this.PrintingResponse === 'function') this.PrintingResponse(m);
     }
 
+    _handleTransactionUpdateMessage(m) 
+    {
+        document.dispatchEvent(new CustomEvent('TxnUpdateMessage', { detail: m }));
```
Comment (mattgresham) [inline src/Spi.js:1461]: The library has two ways of telling a POS that something happened, by dispatching an event or by allowing a method of the library to be overridden (delegate). For this one, we want the second way and to call a method like the _handlePrintingResponse does above.

**Pull request merged**